File: c++/web_tutorial/array2d.cpp
#include <string>
#include "Bridges.h"
#include "Array.h"

using namespace std;
using namespace bridges;

int main(int argc, char **argv) {

	//Bridges::initialize(1, "YOUR_USER_ID", "YOUR_API_KEY");
	if (argc < 3) {
		cout << "Must provide both user id and api key as command line arguments. Terminating.." 
								<< endl;
		exit (-1);
	}
	Bridges *bridges = new Bridges(0, argv[1], argv[2]);
	if (argc == 4)	// Server type provided
		bridges->setServer(argv[3]);
	
	int dims[3] = {4, 4, 1};
	Array<string> *arr = new Array<string>(4, 4);

	for (int i = 0; i < dims[1]; i++)
		for (int j = 0; j < dims[0]; j++)
			arr->getElement(i, j).setLabel("El " + to_string(i * dims[0] + j));

	arr->getElement(0, 0).getVisualizer()->setColor(Color("red"));
	arr->getElement(0, 3).getVisualizer()->setColor(Color("green"));
	arr->getElement(3, 0).getVisualizer()->setColor(Color("blue"));
	arr->getElement(3, 3).getVisualizer()->setColor(Color("magenta"));
	arr->getElement(1, 1).getVisualizer()->setColor(Color("cyan"));
	arr->getElement(2, 2).getVisualizer()->setColor(Color("yellow"));


	Bridges::setTitle("Array Example");
	Bridges::setDataStructure(arr);
	Bridges::visualize();

	return 0;
}

File: python/web_tutorial_mastercopy/array2d.py
from bridges.bridges import *
from bridges.array import *
import sys

def main():
    args = sys.argv[1:]

    # create the Bridges object, set credentials
#if TESTING             
    # command line args provide credentials and server to test on
    bridges = Bridges(int(args[0]), args[1], args[2])
    if len(args) > 3:   
        bridges.connector.set_server(args[3])
#else                   
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
#endif 

    #Bridges title, description 
    bridges.set_title("Two-Dimensional Array Example")
    bridges.set_description("Illustrates a 2D array with visual attributes");


    # for 2D array 6x6
    num_rows = 6
    num_cols = 6
    my_array = Array(y_dim = num_rows, x_dim = num_cols)

    #populate the array
    for row in range(num_rows):
        for col in  range(num_cols):
            my_array[row, col].label = "El " + str(row*num_cols + col);

    #color some of the elements
    my_array[0, 0].color = "red"
    my_array[0, 3].color = "green"
    my_array[5, 0].color = "blue"
    my_array[5, 3].color = "magenta"
    my_array[1, 1].color = "cyan"
    my_array[2, 2].color = "yellow"

    #set visualizer type
    bridges.set_data_structure(my_array)

    #visualize tha array
    bridges.visualize()

if __name__ == "__main__":
    main()

